osu.Game/Screens/OnlinePlay/Lounge/Components/DrawableRoom.cs

```diff
@@ -241,6 +239,9 @@ protected override void LoadComplete()
                 this.FadeInFromZero(transition_duration);
             else
                 Alpha = 0;
+
+            hasPassword.BindTo(Room.HasPassword);
```

Comment:
Hmm, wasn't obvious that this was required (I thought the rooms were recreated each time, but I guess not). Would probably want to add an animation for this, but can be done as a follow up pass.